src/overlays/effects/ovl_Effect_Ss_En_Fire/z_eff_ss_en_fire.c

```diff
@@ -5,23 +5,144 @@
+#define rBodyPart regs[7]
+#define rFlags regs[8]
+#define rScroll regs[9]
 
 #define PARAMS ((EffectSsEnFireInitParams*)initParamsx)
 
-s32 EffectSsEnFire_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+u32 EffectSsEnFire_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsEnFire_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsEnFire_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_En_Fire_InitVars = {
     EFFECT_SS_EN_FIRE,
     EffectSsEnFire_Init,
 };
 
-#endif
+u32 EffectSsEnFire_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsEnFireInitParams* initParams = PARAMS;
+
+    Math_Vec3f_Copy(&this->pos, &initParams->pos);
+    Math_Vec3f_Copy(&this->velocity, &gZeroVec3f);
+    Math_Vec3f_Copy(&this->accel, &gZeroVec3f);
+    this->life = 20;
+    this->rLifespan = this->life;
+    this->actor = initParams->actor;
+    this->rScroll = Rand_ZeroOne() * 20.0f;
+    this->draw = EffectSsEnFire_Draw;
+    this->update = EffectSsEnFire_Update;
+    this->rUnused = -15;
+
+    if (initParams->bodyPart < 0) {
+        this->rYaw = Math_Vec3f_Yaw(&initParams->actor->world.pos, &initParams->pos) - initParams->actor->shape.rot.y;
+        this->rPitch =
+            Math_Vec3f_Pitch(&initParams->actor->world.pos, &initParams->pos) - initParams->actor->shape.rot.x;
+        this->vec.z = Math_Vec3f_DistXYZ(&initParams->pos, &initParams->actor->world.pos);
+    }
+
+    this->rScaleMax = initParams->scale;
+
+    if (initParams->params & ENFIRE_PARAMS_USE_SCALE) {
+        this->rScale = initParams->scale;
+    } else {
+        this->rScale = 0;
+    }
+
+    this->rReg6 = initParams->params & 0x7FFF;
+    this->rBodyPart = initParams->bodyPart;
+    this->rFlags = initParams->flags;
+
+    return 1;
+}
+
+void EffectSsEnFire_Draw(PlayState* play, u32 index, EffectSs* this) {
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+    f32 scale;
+    s16 camYaw;
+    s32 pad[3];
+    s16 redGreen;
+
+    OPEN_DISPS(gfxCtx);
+
+    Matrix_Translate(this->pos.x, this->pos.y, this->pos.z, MTXMODE_NEW);
+    camYaw = (Camera_GetCamDirYaw(GET_ACTIVE_CAM(play)) + 0x8000);
+    Matrix_RotateYS(camYaw, MTXMODE_APPLY);
+
+    scale = Math_SinS(this->life * 0x333) * (this->rScale * 0.00005f);
+    Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    redGreen = this->life - 5;
+
+    if (redGreen < 0) {
+        redGreen = 0;
+    }
+
+    func_8012C2DC(play->state.gfxCtx);
+    gDPSetEnvColor(POLY_XLU_DISP++, redGreen * 12.7f, 0, 0, 0);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0x0, 0x80, redGreen * 12.7f, redGreen * 12.7f, 0, 255);
+    gSPSegment(
+        POLY_XLU_DISP++, 0x08,
+        Gfx_TwoTexScroll(play->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0, (this->rScroll * -20) & 0x1FF, 0x20, 0x80));
```

Comment:
You haven't been very consistent in these between files, but the repo isn't very consistent anyway. Another one to think about later.

Reply:
What's inconsistent?

Reply:
hex vs dec for the fields in the texscrolls

Reply:
Will add to the todo list so it isn't forgotten.